### PR TITLE
refactor: use-snapshot.ts

### PR DESCRIPTION
**Pull request by Donya-qlik**

### CODE REVIEW

File: src/hooks/use-snapshot.ts [hunk 1]
```diff
@@ -12,38 +12,27 @@ interface UseSnapshotProps {
...
-      return copyOfLayout;
+      snapshotLayout.snapshotData.object.size.w = rect.width;
+      snapshotLayout.snapshotData.object.size.h = rect.height;
     }
-
-    return copyOfLayout;
+    return snapshotLayout;
   });
 
   if (layoutService.layout.snapshotData?.content) {
```
Comment (Donya-qlik) [inline src/hooks/use-snapshot.ts:38]: I'm not sure when this condition is used ?! 🤔

Comment (cbt1) [reply src/hooks/use-snapshot.ts:38]: Don't quite remember but could be when the chart is re-sized in storytelling

File: src/hooks/use-snapshot.ts [hunk 2]
```diff
@@ -12,38 +12,27 @@ interface UseSnapshotProps {
 }
 
 const useSnapshot = ({ layoutService, viewService, rect, model }: UseSnapshotProps): stardust.Rect => {
-  onTakeSnapshot(async (copyOfLayout: SnapshotLayout) => {
-    if (!copyOfLayout.snapshotData) {
-      return copyOfLayout;
+  onTakeSnapshot(async (snapshotLayout: SnapshotLayout) => {
+    if (!snapshotLayout.snapshotData || !model || snapshotLayout.snapshotData.content) {
```
Comment (cbt1) [inline src/hooks/use-snapshot.ts:16]: Is the `content` object always there?

Comment (cbt1) [reply src/hooks/use-snapshot.ts:16]: Just did some quick testing and it does seem like there is no `content` the first time `onTakeSnapshot` is called

File: src/hooks/use-snapshot.ts (hunk 1 above)
Comment (cbt1) [reply src/hooks/use-snapshot.ts:38]: Did some testing and cannot find a case where this logic is triggered. For now lets keep it but we might be able to remove it in the feature

File: src/hooks/use-snapshot.ts (hunk 2 above)
Comment (cbt1) [reply src/hooks/use-snapshot.ts:16]: Just had another look. You are indeed correct 👍

**Review by cbt1 — APPROVED** [review]

**Pull request merged**